packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -268,14 +268,28 @@ export function resolveParameter(
   return buildParamEntity(p);
 }
 
-function buildEntityFromParamAndResolveTypeAlias(
+function buildEntityFromParamWithTypeAlias(
   entry: ApiEntryPoint,
   p: Parameter
 ): EntityWithTypeAlias {
   const entity = buildParamEntity(p);
-  const searchableTypeName = extractSearchableTypeName(p.parameterTypeExcerpt);
-  const typeAlias = findApi(entry, searchableTypeName) as ApiTypeAlias;
-  const type = typeAlias.typeExcerpt.text;
+  return tryToResolveTypeAlias(entry, entity, p.parameterTypeExcerpt);
+}
+
+function tryToResolveTypeAlias(
+  entry: ApiEntryPoint,
+  entity: Entity,
+  typeExcerpt: Excerpt
+): EntityWithTypeAlias {
+  const searchableTypeName = extractSearchableTypeName(typeExcerpt);
+  let type: string;
+
+  try {
+    const typeAlias = findApi(entry, searchableTypeName) as ApiTypeAlias;
+    type = typeAlias.typeExcerpt.text;
+  } catch {
+    type = typeExcerpt.text;
+  }
```

Comment:
Resolving a type alias throws an error when the alias is not exported. In this case, I did not want to export `SearchAction` because:
1. It's not a useful type for end-users on it's own.
2. Resolving it would give a long type: `AsyncThunkAction<... not useful generics ...>`.

Rather than throw an error, the type alias is returned unresolved.

I experimented with logging an alert in the console, but I found this too spammy and I think people would eventually just ignore the logs.

Reply:
The thing about the `SearchAction` type is that it's _absolutely necessary_ that the user knows which actions are this type, because they have to know what to pass as a parameter.

The original idea was that all the `SearchAction`s would go under a single `loadAnalyticsSearchActions` function, which I see in the PR description is what you're planning to do, but we then need to be able to refer to those actions as `AnalyticsSearchAction`s or else the user can't make the connection.

I'm not sure if you'd want to keep this code and do some sort of modification afterward to the types of action params, or if you'd prefer to handle it somehow in this code, but it's something to think about.

Reply:
Hmm, yea, I mentioned the action loader in the description to give the reader a cue, but I agree it would be nicer if it was linkable. I'm not sure how we could do that atm though ...

**Utils**

For similar situations with the controllers, we had the concept of `utilities`: functions that go well together with the controller. The utils are not anchor-linked to a controller method though, they are just displayed on the same page.  `executeSearch` is the only one that takes a search action, so it might be logical to place `loadAnalyticsSearchActions` as a util on the same page rather than at the top-level?

**Sample**
Another approach could be to display a code sample on the `loadSearchReducers` page, but again, it would not be linkable and requires a reader to scroll to see it. I think there's a high chance they will though, they've been a success so far.

Maybe a step better, I could include an [@example tag directly in the ts-annotation](https://api-extractor.com/pages/tsdoc/tag_example/)? This would make it possible to show the example right below `executeSearch` instead of at the end of the page. Unlike a code sample, it would also be visible in vscode. We would just need to adjust the doc-parser to extract the tag.

**Url**
Another idea might be to encode a link in the `loadAnalyticsSearchActions` ts-annotation? Not sure if that's feasible though, since the url has a dynamic version number in it? Hard-coding links doesn't seem like a good idea,


What do you think? Any other approaches? I feel a combination of `@example` and either util or top-level documentation of `loadAnalyticsSearchActions` could make things clear,